src/main/java/org/advent/dayten/DayTen.java

```diff
@@ -0,0 +1,70 @@
+package org.advent.dayten;
+
+import org.advent.PrintSolution;
+import org.advent.Util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DayTen implements PrintSolution {
+
+    @Override
+    public void printPartOne() throws IOException {
+        System.out.println(" ");
+        System.out.println(this.getClass().getSimpleName() + " ---------------------------");
+        System.out.println("Number of steps to reach furthest away = " + numberOfStepsToReachFurthestAwayFromS(getStringsFromFile()));
+    }
+
+    @Override
+    public void printPartTwo() throws IOException {
+
+    }
+
+    public List<Position> loopThroughPositions(GridOfPipes grid, Position position, Position previousPosition) {
+        List<Position> listOfPositions = new ArrayList<>(Arrays.asList(previousPosition, position));
+        while (position.character() != 'S') {
+            CoordinateDirection direction = position.nextStep(previousPosition);
+            assert direction != null;
+            char nextCharacter = getNextCharacter(grid, position, direction);
+            previousPosition = position;
+            position = new Position(position.nextX(direction), position.nextY(direction), nextCharacter);
```

Comment:
this part isnt OOP anymore, you started out well but then missed the target.

also this position object already exists, why are you recreating it instead of querying it? don't have access to the function that gives you positions? maybe the function needed has the wrong visibility or this function is in the wrong place. think about it